core/src/main/java/uk/gov/justice/services/core/accesscontrol/AccessControlFailureMessageGenerator.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.services.core.accesscontrol;
+
+import uk.gov.justice.services.messaging.JsonEnvelope;
+import uk.gov.justice.services.messaging.Metadata;
+
+import static java.lang.String.format;
+
+public class AccessControlFailureMessageGenerator {
+
+    public String errorMessageFrom(
+                    JsonEnvelope jsonEnvelope,
+                    AccessControlViolation accessControlViolation) {
+
+        final Metadata metadata = jsonEnvelope.metadata();
+        final String jsonEnvelopeName = metadata.name();
+        final String userId = metadata.userId().orElse("no user id supplied");
+
+        return format("Access Control failed for user id '%s', json envelope '%s'. Reason: %s",
```

Comment:
For trace logging, we have some common components for generating JSON from the metadata of an envelope. We should use that rather than specifically pulling out the user id and name.